[DOCS] Edit validation section of dynamic templates docs (#57510)
elastic/elasticsearch: docs/reference/mapping/dynamic/templates.asciidoc

<2> The match conditions can include any of : `match_mapping_type`, `match`, `match_pattern`, `unmatch`, `path_match`, `path_unmatch`.
<3> The mapping that the matched field should use.

If a provided mapping contains an invalid mapping snippet, a validation error

> will be thrown. Validation occurs when applying the dynamic template at index time, 

Reviewer: ```suggestion
is returned. Validation occurs when applying the dynamic template at index time, 
```

> and in most cases when the dynamic template is updated. Providing an invalid mapping 

Reviewer: ```suggestion
and, in most cases, when the dynamic template is updated. Providing an invalid mapping 
```

> snippet may cause updating or validating the dynamic template to fail under certain conditions:

Reviewer: ```suggestion
snippet may cause the update or validation of a dynamic template to fail under certain conditions:
```

> * If no `match_mapping_type` has been specified, but the template is valid for at least one predefined mapping type, 

Reviewer: ```suggestion
* If no `match_mapping_type` has been specified but the template is valid for at least one predefined mapping type, 
```

>   the mapping snippet is considered valid. However, a validation error will occur at index time if a field matching 

Reviewer: ```suggestion
  the mapping snippet is considered valid. However, a validation error is returned at index time if a field matching 
```

  the template is indexed as a different type. For example, configuring a dynamic template with no `match_mapping_type` 
  is considered valid as string type, but if a field matching the dynamic template is indexed as a long, a validation 

>   error may still occur at index time.

Author: should this read "may" or "will"?

Reviewer: "Will" is correct.

>   error may still occur at index time.

Reviewer: ```suggestion
  error is returned at index time.
```

* If the `{{name}}` placeholder is used in the mapping snippet, validation is skipped when updating the dynamic 

>   template because the field name is unknown at that time. Validation will instead occur when applying the template 

Reviewer: ```suggestion
  template. This is because the field name is unknown at that time. Instead, validation occurs when the template is applied
```
